helper/resource/testing_new_import_state.go

```diff
@@ -238,8 +241,18 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 		}
 
 		// TODO compare plan to state from previous step
+		err = runProviderCommand(ctx, t, func() error {
+			var err error
+			logging.HelperResourceDebug(ctx, "Run terraform apply")
+			err = importWd.Apply(ctx)
+			return err
+		}, importWd, providers)
+		if err != nil {
+			return err
+		}
```

Comment:
🤔 I'm still not sure we should actually apply anything during `ImportState` tests. It feels like an easy way to mess up remote infrastructure during a test without the expectations that a test assertion will point out what went wrong.

The logic above checks the first import plan to be a no-op, but it's still possible for a refresh to change what this apply does with no real feedback to the user.

------

The consequence of not running apply is that we can't use other test steps after an `ImportState`, but maybe that's okay? If the provider developer wants to run an actual apply, they could always use `Config` mode, so maybe that's where we draw the line?

Reply:
I've made these changes based on this feedback:

* Removed the apply logic
* Added support for exactly one flavor of `ImportPlanChecks`: `PreApply`